File: blockchainPAdic/blockchainPAdic.java
import java.util.*;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

class Node {
    String name;
    String hash;
    List<Node> children = new ArrayList<>();

    Node(String name) {
        this.name = name;
    }
}

public class TreeBlockchain {

    static Random rand = new Random();

    static String sha256(String base) {
        try {
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte[] hash = digest.digest(base.getBytes());
            StringBuilder hexString = new StringBuilder();
            for (byte b : hash) {
                String hex = Integer.toHexString(0xff & b);
                if(hex.length() == 1) hexString.append('0');
                hexString.append(hex);
            }
            return hexString.toString();
        } catch(NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
    }

    static String generateName(String level) {
        return level + "_" + (rand.nextInt(1000) + 1);
    }

    static Node generateNode(String level, int depth, String parentHash) {
        Node node = new Node(generateName(level));
        int numChildren = depth > 0 ? rand.nextInt(3) + 1 : 0;

        for(int i=0;i<numChildren;i++) {
            if(depth == 3) node.children.add(generateNode("Bundesland", depth-1, parentHash));
            else if(depth == 2) node.children.add(generateNode("Bezirk", depth-1, parentHash));
            else if(depth == 1) node.children.add(generateNode("Dorf", depth-1, parentHash));
        }

        StringBuilder data = new StringBuilder(node.name + parentHash);
        for(Node c : node.children) data.append(c.hash);
        node.hash = sha256(data.toString());

        return node;
    }

    static void printNode(Node node, int indent) {
        for(int i=0;i<indent;i++) System.out.print("  ");
        System.out.println(node.name + " : " + node.hash);
        for(Node c : node.children) printNode(c, indent+1);
    }

    public static void main(String[] args) {
        Node tree = generateNode("Land", 3, "");
        printNode(tree, 0);
    }
}

File: blockchainPAdic/blockchainPAdic.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>
#include <random>
#include <openssl/sha.h> // für SHA256

struct Node {
    std::string name;
    std::string hash;
    std::vector<Node> children;
};

// Hilfsfunktion: SHA256 Hash aus String
std::string sha256(const std::string &str) {
    unsigned char hash[SHA256_DIGEST_LENGTH];
    SHA256((unsigned char*)str.c_str(), str.size(), hash);
    std::stringstream ss;
    for(int i = 0; i < SHA256_DIGEST_LENGTH; ++i)
        ss << std::hex << (int)hash[i];
    return ss.str();
}

std::string generateName(const std::string &level) {
    return level + "_" + std::to_string(rand() % 1000 + 1);
}

Node generateNode(const std::string &level, int depth, const std::string &parentHash="") {
    Node node;
    node.name = generateName(level);
    int numChildren = (depth > 0) ? (rand() % 3 + 1) : 0;

    for(int i = 0; i < numChildren; ++i) {
        if(depth == 3)
            node.children.push_back(generateNode("Bundesland", depth-1, parentHash));
        else if(depth == 2)
            node.children.push_back(generateNode("Bezirk", depth-1, parentHash));
        else if(depth == 1)
            node.children.push_back(generateNode("Dorf", depth-1, parentHash));
    }

    // Hash berechnen
    std::string data = node.name + parentHash;
    for(const auto &c : node.children)
        data += c.hash;
    node.hash = sha256(data);

    return node;
}

// Hilfsfunktion zum Drucken
void printNode(const Node &node, int indent=0) {
    for(int i=0;i<indent;i++) std::cout << "  ";
    std::cout << node.name << " : " << node.hash << "\n";
    for(const auto &c : node.children)
        printNode(c, indent+1);
}

int main() {
    srand(time(NULL));
    Node tree = generateNode("Land", 3);
    printNode(tree);
    return 0;
}
